components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/view-users.jsp

```diff
@@ -125,37 +125,37 @@
 
     boolean useCache = false;
 
-    if(prevRole != null && prevRole.equals(roleName) ){
+    if (prevRole != null && prevRole.equals(roleName)) {
         useCache = true;
-    } else if(prevRole!=null){
-        session.setAttribute("previousRole",roleName);
+    } else if (prevRole != null) {
+        session.setAttribute("previousRole", roleName);
     }
-    
+
     String readOnlyRoleString = request.getParameter(UserAdminUIConstants.ROLE_READ_ONLY);
-	if (readOnlyRoleString == null) {
-		readOnlyRoleString =
-		                     (String) session.getAttribute(UserAdminUIConstants.ROLE_READ_ONLY);
-	}
-	if ("true".equals(readOnlyRoleString)) {
-		readOnlyRole = true;
-	}
-
-	exceededDomains =
-	                  (FlaggedName) session.getAttribute(UserAdminUIConstants.ROLE_LIST_ASSIGNED_USER_CACHE_EXCEEDED);
-
-	// check page number
-	String pageNumberStr = request.getParameter("pageNumber");
-	if (pageNumberStr == null) {
-		pageNumberStr = "0";
-	}
-
-	try {
-		pageNumber = Integer.parseInt(pageNumberStr);
-	} catch (NumberFormatException ignored) {
-		// page number format exception
-	}
-
-	if(useCache) {
+    if (readOnlyRoleString == null) {
+        readOnlyRoleString =
+                (String) session.getAttribute(UserAdminUIConstants.ROLE_READ_ONLY);
+    }
+    if ("true".equals(readOnlyRoleString)) {
+        readOnlyRole = true;
+    }
```

Comment:
Fixed with #915 
Replacing both if conditions with Boolean.parseboolean will violate the original business logic. Hence applying it for the latter if condition.